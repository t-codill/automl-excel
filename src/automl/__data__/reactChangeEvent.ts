// tslint:disable: no-any
export const reactChangeEvent: React.ChangeEvent<any> = {
    target: {}
} as unknown as React.ChangeEvent<any>;
