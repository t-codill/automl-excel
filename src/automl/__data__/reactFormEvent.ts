// tslint:disable: no-any
export const reactFormEvent: React.FormEvent<any> = {
    preventDefault(): void { return; }
} as unknown as React.FormEvent<any>;
