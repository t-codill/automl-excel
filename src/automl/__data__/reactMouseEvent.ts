// tslint:disable-next-line: no-any
export const reactMouseEvent = {} as unknown as React.MouseEvent<any>;
