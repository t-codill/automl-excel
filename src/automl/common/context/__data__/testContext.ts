import { PageNames } from "../../PageNames";
import { Logger } from "../../utils/logger";
import { IWorkspaceContextProps, WorkspaceError } from "../IWorkspaceProps";
import { WorkspaceFlight } from "../WorkspaceFlight";

export function getLogCustomEventSpy(): jest.SpyInstance<ReturnType<Logger["logCustomEvent"]>> {
    const spy = jest.spyOn(Logger.prototype, "logCustomEvent");
    spy.mockImplementation(jest.fn());
    return spy;
}
let pageName: PageNames = PageNames.Unknown;
export const testContext: IWorkspaceContextProps = {
    get pageName(): PageNames { return pageName; },
    get theme(): string { return "light"; },
    discoverUrls: {},
    flight: new WorkspaceFlight(""),
    location: "eastus",
    resourceGroupName: "testResource",
    subscriptionId: "00000000-0000-0000-0000-000000000000",
    logger: new Logger("test"),
    getToken: () => "testToken",
    setPageName: (value: PageNames) => { pageName = value; },
    workspaceName: "testWorkSpace",
    setNavigationBarButtons(): void { return; },
    onError: (err: WorkspaceError) => { throw err; },
    clearErrors: () => { return; },
    setHeader: () => { return; },
    setLoading: () => { return; }
};
