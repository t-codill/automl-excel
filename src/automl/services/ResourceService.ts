import { ResourceManagementClient, ResourceManagementModels } from "@azure/arm-resources";
import { IServiceBaseProps } from "./ServiceBase";
import { ServiceBaseArm } from "./ServiceBaseArm";

export class ResourceService extends ServiceBaseArm<ResourceManagementClient> {
    constructor(props: IServiceBaseProps) {
        super(props, ResourceManagementClient);
    }

    public async createResourcesGroup(resourceGroupName: string,
        location: string
    ): Promise<ResourceManagementModels.ResourceGroup | undefined> {
        return this.send(async (client, abortSignal) => {
            return client.resourceGroups.createOrUpdate(resourceGroupName, {
                location
            }, { abortSignal });
        });
    }

    public async createWorkspace(
        workspaceName: string,
        resourceGroupName: string,
        location: string
    ): Promise<ResourceManagementModels.DeploymentExtended | undefined> {
        const letterName = workspaceName.replace(/\W/g, "")
            .toLowerCase();
        const keyVaultName = `${letterName}k`;
        const storageName = `${letterName}s`;
        const registriesName = `${letterName}r`;
        const insightsName = `${letterName}i`;
        return this.send(async (client, abortSignal) => {
            return client.deployments.createOrUpdate(resourceGroupName, workspaceName,
                {
                    properties:
                    {
                        template:
                        {
                            $schema: "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
                            contentVersion: "1.0.0.0",
                            parameters: {},
                            variables: {},
                            resources: [
                                {
                                    type: "Microsoft.KeyVault/vaults",
                                    name: keyVaultName,
                                    apiVersion: "2015-06-01",
                                    location,
                                    dependsOn: [],
                                    properties:
                                    {
                                        enabledForDeployment: "true",
                                        enabledForTemplateDeployment: "true",
                                        enabledForVolumeEncryption: "true",
                                        tenantId: "72f988bf-86f1-41af-91ab-2d7cd011db47",
                                        accessPolicies: [],
                                        sku: { name: "Standard", family: "A" }
                                    }
                                },
                                {
                                    type: "Microsoft.Storage/storageAccounts",
                                    name: storageName,
                                    apiVersion: "2016-12-01",
                                    location,
                                    sku: { name: "Standard_LRS" },
                                    kind: "Storage",
                                    dependsOn: [],
                                    properties: {
                                        encryption: {
                                            services: { blob: { enabled: "true" } },
                                            keySource: "Microsoft.Storage"
                                        },
                                        supportsHttpsTrafficOnly: true
                                    }
                                },
                                {
                                    type: "Microsoft.ContainerRegistry/registries",
                                    name: registriesName,
                                    apiVersion: "2017-10-01",
                                    location,
                                    sku: { name: "Standard", tier: "Standard" },
                                    properties: { adminUserEnabled: "true" }
                                },
                                {
                                    type: "microsoft.insights/components",
                                    name: insightsName,
                                    kind: "web",
                                    apiVersion: "2015-05-01",
                                    location: "eastus",
                                    properties: { Application_Type: "web" }
                                },
                                {
                                    type: "Microsoft.MachineLearningServices/workspaces",
                                    name: workspaceName,
                                    apiVersion: "2018-11-19",
                                    identity: { type: "systemAssigned" },
                                    location,
                                    resources: [],
                                    dependsOn: [
                                        `[resourceId('Microsoft.KeyVault/vaults', '${keyVaultName}')]`,
                                        `[resourceId('Microsoft.Storage/storageAccounts', '${storageName}')]`,
                                        `[resourceId('Microsoft.ContainerRegistry/registries', '${registriesName}')]`,
                                        `[resourceId('microsoft.insights/components', '${insightsName}')]`
                                    ],
                                    properties: {
                                        containerRegistry: `/subscriptions/${
                                            this.props.subscriptionId
                                            }/resourceGroups/${
                                            resourceGroupName
                                            }/providers/Microsoft.ContainerRegistry/registries/${registriesName}`,
                                        keyVault: `/subscriptions/${this.props.subscriptionId}/resourceGroups/${resourceGroupName}/providers/Microsoft.KeyVault/vaults/${keyVaultName}`,
                                        applicationInsights: `/subscriptions/${this.props.subscriptionId}/resourceGroups/${resourceGroupName}/providers/microsoft.insights/components/${insightsName}`,
                                        friendlyName: workspaceName,
                                        storageAccount: `/subscriptions/${this.props.subscriptionId}/resourceGroups/${resourceGroupName}/providers/Microsoft.Storage/storageAccounts/${storageName}`
                                    }

                                }]
                        },
                        parameters: {},
                        mode: "Incremental"
                    }
                }
                , { abortSignal });
        });
    }
}
