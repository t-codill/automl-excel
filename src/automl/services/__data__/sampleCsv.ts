export const sampleCsv = `a,b,c,d,e
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
1,2,3,4,5
`;

const largeCsvHeader = Array(50)
    .fill("aaaaa,bbbbb,ccccc,ddddd,eeeee")
    .join(",");

const largeCsvRow = Array(50)
    .fill("11111,22222,33333,44444,55555")
    .join(",");

export const largeCsv = `${largeCsvHeader}
${Array(50)
        .fill(largeCsvRow)
        .join("\n")}
`;
