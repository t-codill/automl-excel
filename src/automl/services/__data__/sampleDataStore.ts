import { AzureMachineLearningDatastoreManagementModels } from "@vienna/datastore";

export const sampleDataStore: Required<AzureMachineLearningDatastoreManagementModels.DataStoreDto> = {
    dataStoreType: "AzureBlob",
    name: "defaultDataStore",
    azureStorageSection: {
        accountKey: "key",
        accountName: "account",
        credentialType: "AccountKey",
        credential: "secret",
        containerName: "defaultContainer",
        isSas: false
    },
    azureDataLakeSection: {},
    azurePostgreSqlSection: {},
    azureSqlDatabaseSection: {},
    hasBeenValidated: true
};
