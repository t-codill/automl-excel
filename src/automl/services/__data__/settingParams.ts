import { AdvancedSetting } from "../AdvancedSetting";

export const settingParams: AdvancedSetting = {
    jobType: "classification",
    column: "1",
    metric: "AUC_weighted",
    experimentTimeoutMinutes: 60,
    maxIteration: 15,
    experimentExitScore: 0.99,
    maxConcurrent: 1,
    maxCores: 2,
    preprocess: true,
    nCrossValidations: 5,
    blacklistAlgos: null,
    validationSize: 0.2,
    grainColumns: null,
    maxHorizon: null,
    timeSeriesColumn: null
};
