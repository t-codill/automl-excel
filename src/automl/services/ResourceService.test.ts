import { testContext } from "../common/context/__data__/testContext";
import { ResourceService } from "./ResourceService";

jest.mock("@azure/arm-resources");

const service = new ResourceService(testContext);

describe("ResourceService", () => {
    it("should createResourcesGroup", async () => {
        const result = await service.createResourcesGroup("sampleResourceGroup", "eastus");
        expect(result)
            .toMatchSnapshot();
    });
    it("should createWorkspace", async () => {
        const result = await service.createWorkspace("sampleWorkSpace", "sampleResourceGroup", "eastus");
        expect(result)
            .toMatchSnapshot();
    });
});
