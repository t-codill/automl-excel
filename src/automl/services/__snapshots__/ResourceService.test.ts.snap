// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceService should createResourcesGroup 1`] = `
Object {
  "_mockParameters": Array [
    "sampleResourceGroup",
    Object {
      "location": "eastus",
    },
    Object {
      "abortSignal": Object {},
    },
  ],
  "location": "eastus",
}
`;

exports[`ResourceService should createWorkspace 1`] = `
Object {
  "_mockParameters": Array [
    "sampleResourceGroup",
    "sampleWorkSpace",
    Object {
      "properties": Object {
        "mode": "Incremental",
        "parameters": Object {},
        "template": Object {
          "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
          "contentVersion": "1.0.0.0",
          "parameters": Object {},
          "resources": Array [
            Object {
              "apiVersion": "2015-06-01",
              "dependsOn": Array [],
              "location": "eastus",
              "name": "sampleworkspacek",
              "properties": Object {
                "accessPolicies": Array [],
                "enabledForDeployment": "true",
                "enabledForTemplateDeployment": "true",
                "enabledForVolumeEncryption": "true",
                "sku": Object {
                  "family": "A",
                  "name": "Standard",
                },
                "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
              },
              "type": "Microsoft.KeyVault/vaults",
            },
            Object {
              "apiVersion": "2016-12-01",
              "dependsOn": Array [],
              "kind": "Storage",
              "location": "eastus",
              "name": "sampleworkspaces",
              "properties": Object {
                "encryption": Object {
                  "keySource": "Microsoft.Storage",
                  "services": Object {
                    "blob": Object {
                      "enabled": "true",
                    },
                  },
                },
                "supportsHttpsTrafficOnly": true,
              },
              "sku": Object {
                "name": "Standard_LRS",
              },
              "type": "Microsoft.Storage/storageAccounts",
            },
            Object {
              "apiVersion": "2017-10-01",
              "location": "eastus",
              "name": "sampleworkspacer",
              "properties": Object {
                "adminUserEnabled": "true",
              },
              "sku": Object {
                "name": "Standard",
                "tier": "Standard",
              },
              "type": "Microsoft.ContainerRegistry/registries",
            },
            Object {
              "apiVersion": "2015-05-01",
              "kind": "web",
              "location": "eastus",
              "name": "sampleworkspacei",
              "properties": Object {
                "Application_Type": "web",
              },
              "type": "microsoft.insights/components",
            },
            Object {
              "apiVersion": "2018-11-19",
              "dependsOn": Array [
                "[resourceId('Microsoft.KeyVault/vaults', 'sampleworkspacek')]",
                "[resourceId('Microsoft.Storage/storageAccounts', 'sampleworkspaces')]",
                "[resourceId('Microsoft.ContainerRegistry/registries', 'sampleworkspacer')]",
                "[resourceId('microsoft.insights/components', 'sampleworkspacei')]",
              ],
              "identity": Object {
                "type": "systemAssigned",
              },
              "location": "eastus",
              "name": "sampleWorkSpace",
              "properties": Object {
                "applicationInsights": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sampleResourceGroup/providers/microsoft.insights/components/sampleworkspacei",
                "containerRegistry": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sampleResourceGroup/providers/Microsoft.ContainerRegistry/registries/sampleworkspacer",
                "friendlyName": "sampleWorkSpace",
                "keyVault": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sampleResourceGroup/providers/Microsoft.KeyVault/vaults/sampleworkspacek",
                "storageAccount": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sampleResourceGroup/providers/Microsoft.Storage/storageAccounts/sampleworkspaces",
              },
              "resources": Array [],
              "type": "Microsoft.MachineLearningServices/workspaces",
            },
          ],
          "variables": Object {},
        },
      },
    },
    Object {
      "abortSignal": Object {},
    },
  ],
  "id": "deploymentId",
  "name": "deploymentName",
}
`;
